tests/backup/backup_restore_basic_test.go

```diff
@@ -1442,3 +1442,220 @@ var _ = Describe("{BackupMultipleNsWithSameLabel}", func() {
+		backupLocationUID        string
+		cloudCredUID             string
+		backupName               string
+		cloudCredUidList         []string
+		appContexts              []*scheduler.Context
+		bkpNamespaces            []string
+		clusterUid               string
+		clusterStatus            api.ClusterInfo_StatusInfo_Status
+		customBackupLocationName string
+		credName                 string
+	)
+	timeBetweenConsecutiveBackups := 4 * time.Second
+	backupNames := make([]string, 0)
+	numberOfSimultaneousBackups := 20
+	labelSelectors := make(map[string]string)
+	bkpNamespaces = make([]string, 0)
+	backupNamespaceMap := make(map[string]string)
+	backupLocationMap := make(map[string]string)
+	credMap := make(map[string]map[string]string)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("CloudSnapsSafeWhenBackupLocationDeleteTest",
+			"Validate if cloud snaps present if backup location is deleted", nil, 58069)
+		log.InfoD("Deploy applications")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Validate that the backup sync syncs all the backups present in bucket", func() {
+		providers := getProviders()
+		Step("Validate applications and get their labels", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
+
+		Step("Adding Credentials and Registering Backup Location", func() {
+			log.InfoD("Using pre-provisioned bucket. Creating cloud credentials and backup location.")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, provider := range providers {
+				cloudCredUID = uuid.New()
+				cloudCredUidList = append(cloudCredUidList, cloudCredUID)
+				backupLocationUID = uuid.New()
+				credName = fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				credMap[provider][cloudCredUID] = credName
+				err := CreateCloudCredential(provider, credName, cloudCredUID, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] with [%s] as provider", credName, orgID, provider))
+				log.InfoD("Created Cloud Credentials with name - %s", credName)
+				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				backupLocationMap[backupLocationUID] = customBackupLocationName
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", customBackupLocationName))
```

Comment:
Specifying the **credential (`credname`) in which backup location was created** is useful info, right? (as we can know which cred a backup location belonged to, in case of failure of this test. The credname will lead up to the provider through the logs).

Reply:
What will you do with this info?

Reply:
We will know the cred name in the logs anyway, just above this line because he is calling out the cred name here.

`log.InfoD("Created Cloud Credentials with name - %s", credName)`

Reply:
I guess that's intuitive enough 👍🏻